src/app/chat/chat-room/chat-room.component.ts

```diff
@@ -1,15 +1,142 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewChild } from "@angular/core";
+import { Router, ActivatedRoute } from "@angular/router";
+import { Content } from "@ionic/angular";
+import { Storage } from "@ionic/storage";
+
+import { ChatService } from "../chat.service";
 
 @Component({
-  selector: 'app-chat-room',
-  templateUrl: './chat-room.component.html',
-  styleUrls: ['./chat-room.component.scss']
+  selector: "app-chat-room",
+  templateUrl: "./chat-room.component.html",
+  styleUrls: ["./chat-room.component.scss"]
 })
 export class ChatRoomComponent implements OnInit {
+  // @TODO need to create method to convert chat time to local time.
+  @ViewChild(Content) content: Content;
+
+  message: any;
+  messageList: any[];
+  selectedChat: any = {
+    name: ""
+  };
+  chatColors: any[];
 
-  constructor() { }
+  constructor(
+    private chatService: ChatService,
+    private router: Router,
+    private storage: Storage
+  ) {}
 
   ngOnInit() {
+    this.loadStorageData();
   }
 
+  loadStorageData() {
+    this.storage.get("chatAvatarColors").then(avatorColors => {
+      this.chatColors = avatorColors;
+      this.storage.get("selectedChatObject").then(chat => {
+        this.selectedChat = chat;
+        this.loadMessages();
+      });
+    });
+  }
+
+  loadMessages() {
+    this.chatService.getMessageList(null).subscribe(response => {
+      this.updateMessageListResponse(response, false);
+    });
+  }
+
+  getChatAvatarText(senderName) {
+    return this.chatService.generateChatAvatarText(senderName);
+  }
+
+  goBack() {
+    this.router.navigateByUrl("/app/(chat:chat)");
+  }
+
+  sendMessage() {
+    if (this.message) {
+      const message = this.message;
+      this.message = "";
+      let data = {
```

Comment:
As we've discussed, don't hardcode in component anymore, build the full logic of component and hardcode data in service only